app/controllers/api/v1/ebikes_controller.rb

```diff
@@ -0,0 +1,42 @@
+class Api::V1::EbikesController < ApplicationController
+  skip_before_action :authenticate_request, only: %i[index]
+
+  def index
+    ebikes = Ebike.where(removed: false).order(created_at: 'desc')
+    render json: ebikes, status: :ok
+  end
+
+  def show
+    ebike = Ebike.find_by(id: params[:id])
+    render json: ebike
+  end
+
+  def create
+    return unless current_user.role.eql?('admin')
+
+    ebike = Ebike.new(ebike_params)
+    ebike.seller_id = current_user.id
+    if ebike.save
+      render json: ebike, status: :ok
+    else
+      render json: { error: 'Ebike not created' }, status: 400
+    end
+  end
+
+  def destroy
+    ebike = Ebike.find_by(id: params[:id])
+    return unless current_user.id.eql?(ebike.seller_id)
+
+    # p('delete params', params)
+    ebike.removed = true
+    ebike.save
+    render json: ebike
+  end
```

Comment:
- [ ] What happens here if the bike is not successfully destroyed?  Kindly add some error handling as you did in the create method (line 14). 👍